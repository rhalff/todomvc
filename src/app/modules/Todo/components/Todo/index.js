// @flow
import React from 'react'
import styles from './scss/Todo.scss'
import TodoHeader from '../TodoHeader/index'
import TodoItems from '../TodoItems/index'
import TodoFooter from '../TodoFooter/index'
import filterItems from '../../lib/filterItems'
import type { Todos } from '../../types/todos'
import type { ItemFilter } from '../../types/filter'

type Props = {
  itemFilter: ItemFilter,
  items: Todos,
  onCreateItem: Function,
  onClearCompleted: Function,
  onFilterClick: Function,
  onToggleComplete: Function,
  onToggleItemComplete: Function,
  onRemoveItem: Function,
  onSaveItem: Function,
  placeholder: string
}

const Todo = ({
  itemFilter,
  items,
  onCreateItem,
  onClearCompleted,
  onFilterClick,
  onToggleComplete,
  onToggleItemComplete,
  onRemoveItem,
  onSaveItem,
  placeholder
}: Props) => {
  let todoItems
  let todoFooter

  if (items.length) {
    todoItems = (
      <TodoItems
        items={filterItems(itemFilter, items)}
        onRemoveItem={onRemoveItem}
        onSaveItem={onSaveItem}
        onToggleItemComplete={onToggleItemComplete}
      />
    )

    todoFooter = (
      <TodoFooter
        filter={itemFilter}
        items={items}
        onClearCompleted={onClearCompleted}
        onFilterClick={onFilterClick}
      />
    )
  }

  return (
    <section className={styles.app}>
      <h1>Todos</h1>
      <TodoHeader
        items={items}
        placeholder={placeholder}
        onCreate={onCreateItem}
        onToggleComplete={onToggleComplete}
      />
      {todoItems}
      {todoFooter}
    </section>
  )
}

export default Todo
