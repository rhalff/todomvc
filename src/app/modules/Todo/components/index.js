export ButtonBar from './ButtonBar'
export Checkbox from './Checkbox'
export ClearCompletedButton from './ClearCompletedButton'
export InputField from './InputField'
export ItemCounter from './ItemCounter'
export LabelEditor from './LabelEditor'
export RemoveButton from './RemoveButton'
export SearchField from './SearchField'
export Todo from './Todo'
export TodoFooter from './TodoFooter'
export TodoHeader from './TodoHeader'
export TodoItem from './TodoItem'
export TodoItems from './TodoItems'
export ToggleComplete from './ToggleComplete'
