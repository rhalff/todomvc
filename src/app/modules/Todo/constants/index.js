// @flow
export { default as FILTER } from './filter'
export { default as TODOS } from './todos'
