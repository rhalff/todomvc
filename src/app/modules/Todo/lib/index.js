// @flow
export { default as allComplete } from './allComplete'
export { default as filterItems } from './filterItems'
