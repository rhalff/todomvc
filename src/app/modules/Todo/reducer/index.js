// @flow
import { combineReducers } from 'redux'
import todos from './todosReducer'
import filter from './filterReducer'

export default combineReducers({
  todos,
  filter
})
