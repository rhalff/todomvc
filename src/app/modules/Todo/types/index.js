// @flow
import type { Store as ReduxStore, Dispatch as ReduxDispatch } from 'redux'

import type { TodosState, TodosAction } from './todos'
import type { FilterState, FilterAction } from './filter'

export type ReduxInitAction = { type: '@@INIT' }

export type State = TodosState & FilterState

export type Action = ReduxInitAction | TodosAction | FilterAction

export type Store = ReduxStore<State, Action>

export type Dispatch = ReduxDispatch<Action>

declare var module: {
  hot: {
    accept(path: string, callback: () => void): void
  }
}

declare var require: any
