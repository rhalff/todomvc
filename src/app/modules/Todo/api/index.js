import { API } from './constants'
import { apiFetch } from '../../../../shared/utils/api'

class TodoApi {
  static getAllTodos(query = {}, fetchingFrom = 'client') {
    return apiFetch(API.TODO.TODOS, { fetchingFrom }, query)
  }
}

export default TodoApi
