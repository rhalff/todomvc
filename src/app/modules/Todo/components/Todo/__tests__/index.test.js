// @flow
import React from 'react'
import Todo from '../index'
import renderer from 'react-test-renderer'

const todos = [
  { id: 1, text: 'Style & compose all components' },
  { id: 2, text: 'App Init', completed: true },
  { id: 3, text: 'Copy Styles one on one' }
]

test('Can show Todo', () => {
  const component = renderer.create(
    <Todo
      itemFilter={() => {}}
      items={todos}
      onCreateItem={() => {}}
      onClearCompleted={() => {}}
      onFilterClick={() => {}}
      onToggleComplete={() => {}}
      onToggleItemComplete={() => {}}
      onRemoveItem={() => {}}
      onSaveItem={() => {}}
      placeholder="What needs to be done?"
    />
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
