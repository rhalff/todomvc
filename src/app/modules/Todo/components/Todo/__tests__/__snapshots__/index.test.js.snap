// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can show Todo 1`] = `
<section
  className="app"
>
  <h1>
    Todos
  </h1>
  <header
    className="todoHeader"
  >
    <input
      checked={false}
      className="unchecked"
      onChange={[Function]}
      type="checkbox"
    />
    <form
      onSubmit={[Function]}
    >
      <input
        className="edit"
        onChange={[Function]}
        placeholder="What needs to be done?"
        type="text"
        value=""
      />
    </form>
    <div />
  </header>
  <section
    className="main"
  >
    <ul
      className="todoItems"
    >
      <li>
        <div
          className="active"
        >
          <input
            checked={undefined}
            className="unchecked"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="labelEditor"
          >
            <div
              contentEditable={false}
              onDoubleClick={[Function]}
              onKeyDown={[Function]}
            >
              Style & compose all components
            </div>
          </div>
          <button
            className="removeButton"
            onClick={[Function]}
          >
            ×
          </button>
        </div>
      </li>
      <li>
        <div
          className="completed"
        >
          <input
            checked={true}
            className="checked"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="labelEditor"
          >
            <div
              contentEditable={false}
              onDoubleClick={[Function]}
              onKeyDown={[Function]}
            >
              App Init
            </div>
          </div>
          <button
            className="removeButton"
            onClick={[Function]}
          >
            ×
          </button>
        </div>
      </li>
      <li>
        <div
          className="active"
        >
          <input
            checked={undefined}
            className="unchecked"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="labelEditor"
          >
            <div
              contentEditable={false}
              onDoubleClick={[Function]}
              onKeyDown={[Function]}
            >
              Copy Styles one on one
            </div>
          </div>
          <button
            className="removeButton"
            onClick={[Function]}
          >
            ×
          </button>
        </div>
      </li>
    </ul>
  </section>
  <footer
    className="todoFooter"
  >
    <div
      className="itemCounter"
    >
      <span>
        2
         
        items
         left
      </span>
    </div>
    <ul
      className="buttonBar"
    >
      <li
        className="listItem"
        onClick={[Function]}
      >
        <a
          className="filterButton"
          href="/#ALL"
          name="ALL"
        >
          All
        </a>
      </li>
      <li
        className="listItem"
        onClick={[Function]}
      >
        <a
          className="filterButton"
          href="/#ACTIVE"
          name="ACTIVE"
        >
          Active
        </a>
      </li>
      <li
        className="listItem"
        onClick={[Function]}
      >
        <a
          className="filterButton"
          href="/#COMPLETED"
          name="COMPLETED"
        >
          Completed
        </a>
      </li>
    </ul>
    <button
      className="clearCompletedButton"
      onClick={[Function]}
    >
      Clear completed
    </button>
  </footer>
</section>
`;
